File: Day-11/Q_43.cpp
//Write a program to Write function to check  prime

#include<iostream>
using namespace std;
void prime(int n){
    
    bool isprime = true;
    int i=2;
    while(isprime && i<n){
        if(n%i==0)
        isprime = false;
        i++;
    }    
    if(isprime)
    cout<<"Prime";
    else
    cout<<"Not Prime";
}

int main(){
    int num;
    cout<<"Enter a number: ";
    cin>>num;
    prime(num);
}

File: Day 11/Q43.java
import java.util.*;
public class Q43
{
    public static boolean prime(int n)
    {
        if(n<=1)
        {
            return false;
        }
        for(int i=2;i<=Math.sqrt(n);i++)
        {
            if(n%i==0)
            {
                return false;
            }
        }
        return true;
    }
    public static void main()
    {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter a number = ");
        int num=sc.nextInt();
        if (prime(num))
        {
            System.out.println(num+" is a prime number");
        }
        else
        {
            System.out.println(num+" is not a prime number");
        }
    }
}
